src/pharmpy/plugins/nlmixr/model.py

```diff
@@ -62,15 +70,35 @@ def convert_model(model):
         description=model.description,
     )
 
-    # Update dataset to lowercase and add evid
-    nlmixr_model = modify_dataset(nlmixr_model)
+    # Update dataset
+    if model.dataset is not None or len(model.dataset) != 0:
+        if keep_etas is True:
+            nlmixr_model = nlmixr_model.replace(
+                modelfit_results=ModelfitResults(
+                    individual_estimates=model.modelfit_results.individual_estimates
+                )
+            )
+        nlmixr_model = translate_nmtran_time(nlmixr_model)
+        # FIXME: dropping columns runs update source which becomes redundant.
+        # drop_dropped_columns(nlmixr_model)
+        if all(x in nlmixr_model.dataset.columns for x in ["RATE", "DUR"]):
+            nlmixr_model = drop_columns(nlmixr_model, ["DUR"])
+        nlmixr_model = nlmixr_model.replace(
+            datainfo=nlmixr_model.datainfo.replace(path=None),
+            dataset=nlmixr_model.dataset.reset_index(drop=True),
+        )
+    else:
+        print_warning("No connected dataset or dataset is empty")
 
-    nlmixr_model = nlmixr_model.update_source()
-    return nlmixr_model
+    # Add evid
+    nlmixr_model = add_evid(nlmixr_model)
 
+    # Check model for warnings regarding data structure or model contents
+    nlmixr_model = check_model(nlmixr_model, skip_error_model_check=skip_check)
 
-def name_mangle(s):
-    return s.replace('(', '').replace(')', '').replace(',', '_')
+    nlmixr_model.update_source()
+
+    return nlmixr_model
 
 
 class ExpressionPrinter(sympy_printing.str.StrPrinter):
```

Comment:
Wasn't the `ExpressionPrinter` moved to another module or are there two printers now?

Reply:
Correct. This has now been removed as it should have been and is only defined in ìni.py`